api/src/main/java/org/openmrs/module/fhir2/api/translators/impl/DosageTranslatorImpl.java

```diff
@@ -73,6 +80,11 @@ public Dosage toFhirResource(@Nonnull DrugOrder drugOrder) {
 	@Override
 	public DrugOrder toOpenmrsType(@Nonnull DrugOrder drugOrder, @Nonnull Dosage dosage) {
 		drugOrder.setDosingInstructions(dosage.getText());
+		
+		if (!dosage.getAdditionalInstruction().isEmpty()) {
```

Comment:
Dumbly, `get...()` in the FHIR methods always creates an empty object. `has...()` won't create the object unnecessarily.

```suggestion
		if (!dosage.hasAdditionalInstructions()) {
```